### PR TITLE
Implement FORMAT()

### PR DESCRIPTION
**Pull request by rachithrr**
This works the same way as the print function works.

### CODE REVIEW

**Review by ljsinclair — APPROVED** [review]
looks good to me :)

**Review by gthrone25 — CHANGES_REQUESTED** [review]
Is it possible to get  some more examples? ie Boolean and string relationship might be nice to see what the string representation is. Are there any known combinations that won't work? Can you use IDSET and STRINGSETs with these?

**Review by ljsinclair — COMMENTED** [review]
Molto bene! Makeit so I say :D

**Review by ljsinclair — CHANGES_REQUESTED** [review]
Can I just ask now I've really looked at the SQL whether we can also define the additional arguments in the syntax, e.g., 

`"%s -> #%x"` and "`(`id=%d, time =%v...` etc

These may be wildcards (?) that can be used elsewhere too, so if that's the case, I'll set them up as reusable content then insert them where needed. I will need guidance on this though :D

File: sql-preview/functions/_posts/2023-01-26-sql-format.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+A. Format with multiple arguments.
+
+```sql
+create table segments
+    (_id id, segment string, value int);
+
+insert into segments(_id, segment, value)
+    values (1,'white', 16777215);
+
+select _id, format("%s -> #%x", segment, value) as segment from segments;
```
Comment (ljsinclair) [inline sql-preview/functions/_posts/2023-01-26-sql-format.md:52]: Here's the select I'm talking about in my earlier comment.

There's things that aren't defined in the Syntax that shuld be so users know how to use the function.
And as I say, if the `-> #%x` is some kind of known wildcard or prefix or whatnot, then they need to be clearly defined somewhere in the docs, even if it's a hyperlink out to somewhere like w3schools where they're fully defined.

File: sql-preview/functions/_posts/2023-01-26-sql-format.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+B. Format with default values and sets.
+```sql
+create table segments
+    (_id id, time timestamp timeunit 'ms' epoch '2022-01-01T00:00:00Z', ids idset, strings stringset);
+
+insert into segments(_id, time, ids, strings)
+    values (1, '2023-01-01', [6 , 1, 9], ['red', 'blue', 'green']);
+
+select format('id = %d , time = %v , ids = %d, strings = %s', _id, time, ids, strings) as description from segments;
```
Comment (ljsinclair) [inline sql-preview/functions/_posts/2023-01-26-sql-format.md:68]: Here's the other instance of the "not quite clear syntax running into actual examples" :D

File: sql-preview/functions/_posts/2023-01-26-sql-format.md (hunk 1 above)
Comment (rachithrr) [reply sql-preview/functions/_posts/2023-01-26-sql-format.md:52]: The %x is the standard format specifier for hexadecimal, as i have mentioned in the # remarks section. This function is similar to the printf() in C. It wraps around fmt.Sprintf(). 
This is a limited for now. We will add more functionalities like 'currency', 'date' etc. in  the future. 


The '-> #' part is just the string that gets printed as is. %x refers to hexadecimal. Similarly, %d for int and %s for strings.

Comment (gthrone25) [reply sql-preview/functions/_posts/2023-01-26-sql-format.md:52]: Agree with @rachithrr, those are standard and he's defined them above. I think this is great!

File: sql-preview/functions/_posts/2023-01-26-sql-format.md (hunk 2 above)
Comment (gthrone25) [reply sql-preview/functions/_posts/2023-01-26-sql-format.md:68]: Sorry Lisa, gonna have to gang up on you with @rachithrr  here as these are pretty standard. Thanks for adding this example. This helps highlight weird types for users! One q, we are sure the commas don't separate values? That could get very weird if spaces exist in values for Stringsets

File: sql-preview/functions/_posts/2023-01-26-sql-format.md (hunk 1 above)
Comment (ljsinclair) [reply sql-preview/functions/_posts/2023-01-26-sql-format.md:52]: No worries :D

File: sql-preview/functions/_posts/2023-01-26-sql-format.md (hunk 2 above)
Comment (ljsinclair) [reply sql-preview/functions/_posts/2023-01-26-sql-format.md:68]: No worries :D

File: sql-preview/functions/_posts/2023-01-26-sql-format.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
++-------------------------------------------------------------------------------------------+
+| description                                                                               |
++-------------------------------------------------------------------------------------------+
+| id = 1 , time = 2023-01-01 00:00:00 +0000 UTC , ids = [1 6 9], strings = [blue green red] |
```
Comment (gthrone25) [inline sql-preview/functions/_posts/2023-01-26-sql-format.md:72]: @rachithrr last Q, but you are sure commas don't return with sets? Seems weird if data had spaces in them, but just looking to confirm.

Comment (rachithrr) [reply sql-preview/functions/_posts/2023-01-26-sql-format.md:72]: Nope. All these are run in the cli and pasted the outputs as is.

[the remark this answers is not in the document]
Comment (ljsinclair) [reply sql-preview/functions/_posts/2023-01-26-sql-format.md:72]: FYI, on further testing, the time jump is caused by the INSERT INTO statement.
```
INSERT INTO segments(_id, time, ids, strings)
    VALUES (1, '2023-01-01', [6 , 1, 9], ['red', 'blue', 'green']);
```

Comment (ljsinclair) [reply sql-preview/functions/_posts/2023-01-26-sql-format.md:72]: FYI, running the second SQL example has revealed a time-travel bug:
https://github.com/molecula/documentation/pull/315/files#

**Comment by ljsinclair** [comment]
FYI this content added to https://github.com/FeatureBaseDB/featurebase-docs/tree/new-functions and will be merged when this is approved/merged

File: sql-preview/functions/_posts/2023-01-26-sql-format.md (hunk 3 above)
Comment (rachithrr) [reply sql-preview/functions/_posts/2023-01-26-sql-format.md:72]: yeah, I asked @paddyjok about it. It's a known bug with epochs that is not related to FORMAT(). It should be fixed soon.  
Note that you get the same `3992` on `select`  statement as well.

Comment (ljsinclair) [reply sql-preview/functions/_posts/2023-01-26-sql-format.md:72]: Yeah, it's pretty impressive :D

**Review by gthrone25 — APPROVED** [review]
Approved!

**Comment by gthrone25** [comment]
@ljsinclair need you to approve as you "requested changes"

**Review by ljsinclair — APPROVED** [review]

**Pull request merged**